src/xcode/ENA/ENA/Source/Coordinator.swift

```diff
@@ -0,0 +1,282 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import UIKit
+import Connectivity
+
+protocol CoordinatorDelegate: AnyObject {
+	func coordinatorUserDidRequestReset()
+}
+
+class Coordinator: RequiresAppDependencies {
+	private weak var delegate: CoordinatorDelegate?
+
+	private let rootViewController: UINavigationController
+
+	private var homeController: HomeViewController?
+	private var settingsController: SettingsViewController?
+	private var exposureDetectionController: ExposureDetectionViewController?
+
+	private lazy var exposureSubmissionService: ExposureSubmissionService = {
+		ExposureSubmissionServiceFactory.create(
+			diagnosiskeyRetrieval: self.exposureManager,
+			client: self.client,
+			store: self.store
+		)
+	}()
+
+	private var enStateUpdatingSet = NSHashTable<AnyObject>.weakObjects()
+
+	init(_ delegate: CoordinatorDelegate, _ rootViewController: UINavigationController) {
+		self.delegate = delegate
+		self.rootViewController = rootViewController
+	}
+
+	deinit {
+		enStateUpdatingSet.removeAllObjects()
+	}
+
+	func showHome(enStateHandler: ENStateHandler, state: SceneDelegate.State) {
+		let vc = AppStoryboard.home.initiate(viewControllerType: HomeViewController.self) { [unowned self] coder in
+			HomeViewController(
+				coder: coder,
+				delegate: self,
+				detectionMode: state.detectionMode,
+				exposureManagerState: state.exposureManager,
+				initialEnState: enStateHandler.state,
+				risk: state.risk,
+				exposureSubmissionService: self.exposureSubmissionService
+			)
+		}
+
+		homeController = vc // strong ref needed
```

Comment:
is there reason to do it like this,
```swift
 homeController = vc 
```

Reply:
I think, it was just copied from app/scene delegate.

Reply:
so we need to keep reference to `homeViewController` in order to update it later

Reply:
Aleksandr is right. It was just copied from ```SceneDelegate```.